quipucords/scanner/network/connect.py

```diff
@@ -301,8 +302,12 @@ def _connect(  # noqa: PLR0913, PLR0912, PLR0915
                 verbosity=verbosity_lvl,
             )
         except Exception as err_msg:  # noqa: BLE001
+            delete_ssh_keyfiles(inventory)
             raise AnsibleRunnerException(err_msg) from err_msg
 
+        # Let's delete any private ssh key files that we generated
+        delete_ssh_keyfiles(inventory)
```

Comment:
I'm not thrilled about the `delete_ssh_keyfiles` being explicitly called in multiple places. 😕  I wish we had a context manager that could do file cleanup automatically, but it looks like the loading and accessing of this data is so complexly intertwined (in legacy code, not just this new code), after walking through code for a few minutes, I couldn't find anywhere reasonably "clean" that could be wrapped like that.

I'll accept this since any alternative that I currently see would be _a lot_ more work, but I'm putting this here to see if anyone else has similar feelings or alternate suggestions

Reply:
I know, I didn't like it either 😢 , luckily it was only the two places.